test/tests/brakeman.rb

```diff
@@ -29,6 +29,42 @@ def test_relative_path_in_warnings
 
 
   end
+
+  def test_app_tree_flexible_file_paths
+    require 'brakeman/options'
+    relative_path = File.expand_path(File.join(File.dirname(__FILE__), "..", "/apps/rails4_non_standard_structure/"))
+    input = ["-p", relative_path.to_s]
+    options, _ = Brakeman::Options.parse input
+    at = Brakeman::AppTree.from_options options
+
+    contains_foo_controller = false
+    at.controller_paths.each do |path|
+      if File.basename(path) == 'foo.rb'
+        contains_foo_controller = true
+        break
+      end
+    end
+
+    contains_foo_model = false
+    at.model_paths.each do |path|
+      if File.basename(path) == 'foo.rb'
```

Comment:
These file names should probably be different so it is clear the test is working correctly.